Modified the way to add singleton resources.

olalonde opened the pull request:
See README.md. I think it's much cleaner now. Mapping the singleton resource to a controller shouldn't be railway-routes' job since it's not expected as you mentioned. Let me know if you like this version better. I hope I didn't forget to follow the conventions too many times :/

olalonde committed:
Singleton resources can now have nested resources.

olalonde committed:
Modified the way to add singleton resources. Singleton resources can now have nested resources.

anatoliychakkaev commented on examples/singleton/server.js:
```diff
@@ -0,0 +1,29 @@
+});
+
+map.resources('users', function(user) {
+  map.resources('posts', function(post) {
```
I guess it should be `user.resources('posts',...`?

olalonde replied:
Oops yeah :/

olalonde committed:
When using the path parameter, nested subroutes should be prefixed by the path, not the resource's name

olalonde commented:
Hey I fixed the README and added another commit to implement something I needed. Let me explain what I'm trying to achieve first. In my app, there are 3 ways to access a user's profile.

/account (user profile of currently logged user)
/users/:user_id (any user's profile using user_id)
/:username (any user's profile using username)

All those routes should have the same nested resources (subroutes) as they all basically represent the same resource. Here's how I implemented it:

```
var addUserSubresources = function(user) {
  user.resources('posts');
};

map.resources('users', addUserSubresources);
map.resource('account', addUserSubresources);
map.resource('users' , {path: ':username'}, addUserSubresources);
```

ps: thinking about it now, I should actually replace `map.resource('account', addUserSubresources);` by `map.resource('users', {path: 'account'}, addUserSubresources);`

Anyways, the problem was that the nested routes path always begun with /users/. So I think I fixed this in this commit by prefixing subroutes by params.path if it is available, otherwise default to resource name like before.

It would also be nice if I could do something like below and I may work on implement it if you like the idea.

```
map.resources('users', {paths: [':username', 'account']}, function(user) {
  user.resources('posts');
});
```

Let me know what you think about this.

olalonde commented:
Oops, forgot to remove my console.log()-foo :)

olalonde committed:
Removed console.log garbage.
